File: csharp/AppEncryption/AppEncryption.IntegrationTests/Utils/DateTimeUtils.cs
using System;

namespace GoDaddy.Asherah.AppEncryption.IntegrationTests.Utils
{
    public static class DateTimeUtils
    {
        /// <summary>
        /// Returns current UTC time in ISO 8601 format
        /// </summary>
        /// <returns>A string which contains the Datetime in ISO 8601 format</returns>
        public static string GetCurrentTimeAsUtcIsoDateTimeOffset()
        {
            return DateTimeOffset.UtcNow.ToString("o");
        }

        /// <summary>
        /// Returns a datetime converted to ISO 8601 format
        /// </summary>
        /// <param name="dateTimeOffset">A DateTimeOffset object that need to be converted to ISO 8601 format</param>
        /// <returns>A string which contains the input DateTime in ISO 8601 format</returns>
        public static string GetDateTimeAsUtcIsoOffsetDateTime(DateTimeOffset dateTimeOffset)
        {
            return dateTimeOffset.ToString("o");
        }
    }
}

File: java/app-encryption/src/it/java/com/godaddy/asherah/utils/DateTimeUtils.java
package com.godaddy.asherah.utils;

import java.time.Instant;
import java.time.OffsetDateTime;
import java.time.ZoneOffset;
import java.time.format.DateTimeFormatter;

public final class DateTimeUtils {
  private DateTimeUtils() {
  }

  /**
   * Returns current UTC time as {@link java.time.format.DateTimeFormatter#ISO_DATE}.
   * @return
   */
  public static String getCurrentTimeAsUtcIsoDate() {
    return OffsetDateTime.now(ZoneOffset.UTC).format(DateTimeFormatter.ISO_DATE);
  }

  /**
   * Returns current UTC time as {@link java.time.format.DateTimeFormatter#ISO_OFFSET_DATE_TIME}.
   * @return
   */
  public static String getCurrentTimeAsUtcIsoOffsetDateTime() {
    return OffsetDateTime.now(ZoneOffset.UTC).format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
  }

  /**
   * Returns the UTC time as {@link java.time.format.DateTimeFormatter#ISO_OFFSET_DATE_TIME} for the given {@code instant}.
   * @param instant
   * @return
   */
  public static String getInstantAsUtcIsoOffsetDateTime(final Instant instant) {
    return OffsetDateTime.ofInstant(instant, ZoneOffset.UTC).format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
  }
}
